dashboard/lib/services/lti.rb

```diff
@@ -160,31 +171,53 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_members)
       had_changes = false
       client_id = lti_integration.client_id
       issuer = lti_integration.issuer
-      current_students = nrps_members.map do |nrps_member|
-        student = Queries::Lti.get_user_from_nrps(
+      section = lti_section.section
+      current_students = Set.new
+      current_teachers = Set.new
+
+      nrps_members.each do |nrps_member|
+        account_type = Policies::Lti.get_account_type(nrps_member[:roles])
+
+        user = Queries::Lti.get_user_from_nrps(
           client_id: client_id,
           issuer: issuer,
           nrps_member: nrps_member
         )
-        student ||= initialize_lti_student_from_nrps(
+        user ||= initialize_lti_user_from_nrps(
           client_id: client_id,
           issuer: issuer,
           nrps_member: nrps_member
         )
-        had_changes ||= (student.new_record? || student.changed?)
-        student.save!
-        add_student_result = lti_section.section.add_student(student)
-        had_changes ||= add_student_result == Section::ADD_STUDENT_SUCCESS
-        student
+        had_changes ||= (user.new_record? || user.changed?)
+        user.save!
+        if account_type == ::User::TYPE_STUDENT
```

Comment:
Nit: elsewhere in the code the condition we're checking for is `User::TYPE_TEACHER` and letting everything that doesn't fit that criteria fall to the `else` block which assumes the user is a student. Would it make sense to follow that pattern here to consistency?